### PR TITLE
Added functionality to the ArcgisItemCard 

### PR DESCRIPTION
**Pull request by jcoma13**
Updated functionality added to the ArcgisItemCard (vertical version, both with and without actions, as well as a horizontal card with actions)

## Description
This addition to the ArcgisItemCard allows users to have more options for how they wish to use the ArcgisItemCard. The vertical card gives users more options for how they would like to display the card that they would not have been able to do with the original card. Additionally, adding the ability to put actions on both the horizontal and vertical versions of the card creates a more diverse functionality for the ArcgisItemCard as a whole.

## Motivation and Context
This change adds functionality to the ArcgisItemCard.

## How Has This Been Tested?
I tested these changes by running the code and seeing the changes that were made to the Calcite React page until I was satisfied with how the modified component looked.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.

### CODE REVIEW

File: src/ArcgisItemCard/ArcgisItemCard-styled.js [hunk 1]
```diff
@@ -71,14 +87,20 @@ StyledItemCardImageWrap.defaultProps = { theme };
 
 const StyledCardItemTitle = styled(StyledCardTitle)`
   ${fontSize(1)};
-  margin-bottom: ${props => unitCalc(props.theme.baseline, 5, '/')};
+  margin-top: ${props => unitCalc(props.theme.baseline, 5, '/')};
```
Comment (codylawson) [inline src/ArcgisItemCard/ArcgisItemCard-styled.js:90]: The margin looks off here, and when I compare against the card in the already deployed version. Can you make sure the styles stay consistent with the latest docs

File: src/ArcgisItemCard/ArcgisItemCard-styled.js [hunk 2]
```diff
@@ -90,7 +112,10 @@ const StyledCardItemText = styled.p`
   font-weight: 300;
   font-style: normal;
   ${fontSize(-2)};
-  margin: ${props => unitCalc(props.theme.baseline, 3, '/')} 0 0;
+  margin-top: ${props => unitCalc(props.theme.baseline, 3, '/')};
```
Comment (codylawson) [inline src/ArcgisItemCard/ArcgisItemCard-styled.js:115]: When defining all margin directions we typically use the shorthand:
```
margin: ${props => unitCalc(props.theme.baseline, 3, '/')};
```

File: src/ArcgisItemCard/ArcgisItemCard.js [hunk 3]
```diff
@@ -112,12 +119,15 @@ ArcgisItemCard.propTypes = {
   /** AGOL portal object - if not specified will default to ArcGIS Online */
   portal: PropTypes.object,
   /** AGOL login token. */
-  token: PropTypes.string
+  token: PropTypes.string,
+  /** Style prop to position Card content vertically */
+  vertical: PropTypes.bool
```
Comment (codylawson) [inline src/ArcgisItemCard/ArcgisItemCard.js:124]: We'll need to add the `actions` prop type and comment here.

File: src/Card/Card-styled.js [hunk 4]
```diff
@@ -114,16 +114,25 @@ const StyledCardImage = styled.img`
...
-    css`
-      width: auto;
-      min-width: 100%;
-      max-width: none;
-      height: 100%;
-      margin: 0;
-      display: block;
-      position: absolute;
-    `};
+    props.wide
```
Comment (codylawson) [inline src/Card/Card-styled.js:117]: Styles that are not conditional based on a prop should be defined above the props arrow function, eg line 115

File: src/ArcgisItemCard/ArcgisItemCard-styled.js (hunk 1 above)
Comment (jcoma13) [reply src/ArcgisItemCard/ArcgisItemCard-styled.js:90]: Would you recommend using padding here instead? Also, where would I need to go to find the styles in the already deployed version? Just within the other components?

Comment (codylawson) [reply src/ArcgisItemCard/ArcgisItemCard-styled.js:90]: Nah I think margin is fine.

You could just visually compare agains the deployed app at https://calcite-react.netlify.app/ . If you need to see the code you can also look at the component files on github in the develop branch.

**Pull request merged**